nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/ConsumeMQTT.java

```diff
@@ -440,15 +440,7 @@ public void process(final OutputStream out) throws IOException {
 
             session.getProvenanceReporter().receive(messageFlowfile, getTransitUri(mqttMessage.getTopic()));
             session.transfer(messageFlowfile, REL_MESSAGE);
-            session.commit();
-            if (!mqttQueue.remove(mqttMessage) && logger.isWarnEnabled()) {
```

Comment:
No more warn message?

Reply:
No, it didn't make sense - given that processor is @TriggerSerially it can only have single thread operating here at at a time. So the call to remove will always return true. And if it didn't - the message doesn't really make any sense. It expects that the person seeing the message understands the code behind the processor.